From d35281f42af752b1b6634262f05d669d257df6a2 Mon Sep 17 00:00:00 2001
From: Joshua Rodriguez <ewized@gmail.com>
Date: Mon, 24 Nov 2014 16:30:52 -0800
Subject: [PATCH] Disable tab display names untill proper api


diff --git a/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java b/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
index 9486772..b31b7de 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
@@ -132,6 +132,8 @@ public class PacketPlayOutPlayerInfo extends Packet {
             packetdataserializer.b( this.entries.size() );
 
             for(Entry entry : this.entries) {
+                entry.displayName = null; // FireCast Disable display names until proper api
+                boolean displayName = entry.displayName != null;
                 packetdataserializer.writeUUID( entry.uuid );
                 switch ( action )
                 {
@@ -150,8 +152,8 @@ public class PacketPlayOutPlayerInfo extends Packet {
                         }
                         packetdataserializer.b( entry.gamemode );
                         packetdataserializer.b( entry.ping );
-                        packetdataserializer.writeBoolean( entry.displayName != null );
-                        if ( entry.displayName != null )
+                        packetdataserializer.writeBoolean( displayName );
+                        if ( displayName )
                         {
                             packetdataserializer.a(entry.displayName);
                         }
@@ -163,8 +165,8 @@ public class PacketPlayOutPlayerInfo extends Packet {
                         packetdataserializer.b( entry.ping );
                         break;
                     case UPDATE_DISPLAY_NAME:
-                        packetdataserializer.writeBoolean( entry.displayName != null );
-                        if ( entry.displayName != null )
+                        packetdataserializer.writeBoolean( displayName );
+                        if ( displayName )
                         {
                             packetdataserializer.a(entry.displayName);
                         }
-- 
1.9.1

