From c5ec7f51700fd8ac5c7f651e0d5024c6c6c71000 Mon Sep 17 00:00:00 2001
From: Joshua Rodriguez <ewized@gmail.com>
Date: Thu, 20 Nov 2014 07:29:25 -0800
Subject: [PATCH] Run onDisable before setting the plugin to disable


diff --git a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
index 19893f3..4c62bc4 100644
--- a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
+++ b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
@@ -309,13 +309,14 @@ public abstract class JavaPlugin extends PluginBase {
      * @param enabled true if enabled, otherwise false
      */
     protected final void setEnabled(final boolean enabled) {
-        if (isEnabled != enabled) {
-            isEnabled = enabled;
-
-            if (isEnabled) {
-                onEnable();
-            } else {
+        if(!isEnabled && enabled) {
+            isEnabled = true;
+            onEnable();
+        } else if(isEnabled && !enabled) {
+            try {
                 onDisable();
+            } finally {
+                isEnabled = false;
             }
         }
     }
-- 
1.9.1

